pennylane/measurements/probs.py

```diff
@@ -264,6 +264,9 @@ def process_density_matrix(self, density_matrix: TensorLike, wire_order: Wires):
             )
 
         # Since we only care about the probabilities, we can simplify the task here by creating a 'pseudo-state' to carry the diagonal elements and reuse the process_state method
+        # Introduce a small epsilon to avoid sqrt(0) to ensure absolute positivity.
+        prob = qml.math.real(prob)
+        prob = qml.math.clip(prob, 1e-14, 1.0)
```

Comment:
```suggestion
        prob = qml.math.where(prob==0, prob+1e-14, prob)
```
It turned out always shifting by 1e-14 casues issues for other corner cases: https://github.com/PennyLaneAI/pennylane/actions/runs/12363310789/job/34505082258?pr=6684